public/js/selfoss-db.js

```diff
@@ -1,6 +1,154 @@
 /**
  * db functions: client data repository
  */
+
+
+selfoss.dbOnline = {
```

Comment:
Was there any particular motivation for moving it into a separate file?

The main issue I have with this is probably the entangledness of the architecture, though refactoring it is a long term goal.